app/components/shared/inputs/BoolInput.tsx

```diff
@@ -0,0 +1,23 @@
+import { Component, Prop } from 'vue-property-decorator';
+import { BaseInput } from './BaseInput';
+
+@Component({})
+export default class BoolInput extends BaseInput<boolean, { name: string; title: string }> {
+  @Prop() readonly value: boolean;
+  @Prop() readonly title: string;
```

Comment:
Is there anything else required to get type checking for these props?  I thought there was more boilerplate involved here.

Reply:
nope! `BaseInput` is where the boilerplate exists in that regard. boilerplate will always live on the (formerly) `extends Vue` layer in the inheritance chain, but once that is defined you can inherit off of it just fine.